L95-python-reflection-client.md

```diff
@@ -0,0 +1,60 @@
+Python Reflection Client
+----
+* Author: Tomer Vromen
+* Approver: a11r, lidizheng (proposed)
+* Status: Draft
+* Implemented in: Python
+* Last updated: 2022-01-20
+* Discussion at: (TBD, filled after thread exists)
+
+## Abstract
+
+Add a programmatic way to access reflection descriptors from client-side Python.
+
+## Background
+
+Reflection is a means by which a server can describe which services and messages it supports.
+The Reflection protocol is already part of the [gRPC repository](https://github.com/grpc/grpc/blob/master/doc/server-reflection.md),
+and there are server-side implementations of it in several languages, including C++ and Python - see [gRPC Python Server Reflection](https://github.com/grpc/grpc/blob/master/doc/python/server_reflection.md)
+
+However, the only client-side implementation is in [C++](https://github.com/grpc/grpc/blob/master/doc/server_reflection_tutorial.md#use-server-reflection-in-a-c-client).
+It seems that a client-side Python implementation is a missing to complete the picture.
+
+This proposal closes this gap.
+
+### Related Proposals: 
+
+* Loosely related: [Promote the Reflection Service from v1alpha to v1](https://github.com/grpc/proposal/blob/master/A15-promote-reflection.md)
+    - The proposal discusses the stability of the reflection service.
+    - Accepted but not yet implemented - see [#27957](https://github.com/grpc/grpc/pull/27957).
+
+## Proposal
+
+Provide a Python implementation for client-side reflection, modeled after the existing C++ implementation.
+
+* Implement `ProtoReflectionDescriptorDatabase` which implements the
+[`DescriptorDatabase`](https://googleapis.dev/python/protobuf/latest/google/protobuf/descriptor_database.html#google.protobuf.descriptor_database.DescriptorDatabase)
+interface.
+* Write tests.
+* Write documentation.
```

Comment:
You may want to update https://github.com/grpc/grpc/blob/master/doc/python/sphinx/grpc_reflection.rst to add the new APIs.

Reply:
good idea, will do that

Reply:
Done.